submissions/nataliaArtsiukh/Frogger-game/app.js

```diff
@@ -0,0 +1,141 @@
+var colWidth = 101;
+var rowHeight = 83;
+
+var Star = function(numCol) {
+    this.x = colWidth * numCol;
+    this.y = -10;
+    this.sprite = 'images/Star.png';
+}
+
+Star.prototype.update = function() {
+    if (player.y < 0) {
+        var index = player.x / 101;
+        allStars[index] = null;
+        if (allStars.filter(elem => elem).length == 0) {
+            setTimeout(()=> {
+                alert('You are amazing!!!'); 
+                launch();
+            }, 1000);
+        } 
+    }
+}
+
+Star.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+}
+
+// Enemies our player must avoid
+
+var Enemy = function(numRow) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.x = colWidth * -1;
+    this.y = rowHeight * numRow - 20;
+    this.speed = (Math.random() + 0.7) * colWidth;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    if (this.x < colWidth * 5) {
+        this.x = this.x + this.speed * dt;
+    } else {
+        this.x = colWidth * -1;
+        this.speed = (Math.random() + 0.7) * colWidth;
+    }
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+var Player = function() {
+    this.x = colWidth * 2;
+    this.y = rowHeight * 5 - 10;
+    this.finished = false;
+    this.sprite = 'images/char-boy.png';
+    this.update = function(){
```

Comment:
In JS prototypical class model methods are properly declared in a different way.
Check [this MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain). Feel free to find other materials that would explain the model better.

BTW, how happened methods for Star and Enemy are defined with proper syntax?

Reply:
I assumed that since the player exists in a single instance, the prototype is not needed. Now everything is fixed.